src/test/java/com/wikia/webdriver/testcases/discussions/ReportingPostTests.java

```diff
@@ -0,0 +1,620 @@
+  }
+
+  @Test(groups = "discussions-loggedInUsersDesktopReporting",
+      dependsOnMethods = {
+          "userOnDesktopCanReportPostOnPostDetailsPage",
+          "anonUserOnDesktopCanNotSeeReportedPostOnPostDetailsPage",
+          "userOnDesktopCannotSeeReportedIndicatorOnPostsReportedByAnotherUserOnPostDetailsPageAndCanReportThatPost",
+          "moderatorOnDesktopCanApproveReportedPostOnPostDetailsPage"})
+  @Execute(asUser = User.USER)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
+  public void userOnDesktopCannotReReportPostOnPostDetailsPage() {
+    final String postId = testsPostData.get(POST_ON_DESKTOP_ON_POST_DETAILS).getId();
+
+    PostDetailsPage page = new PostDetailsPage().open(postId);
+
+    PostEntity postEntity = page.getPost().findPostById(postId);
+
+    assertThatPostIsNotReported(postEntity);
+    assertThatReportPostOptionIsNotAvailableFor(postEntity);
+  }
+
+  // Second user on desktop
+
+  @Test(groups = "discussions-loggedInUsersDesktopReporting",
+      dependsOnMethods = "userOnDesktopCanReportPostOnPostsListPage")
+  @Execute(asUser = User.USER_2)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
+  public void userOnDesktopCannotSeeReportedIndicatorOnPostsReportedByAnotherUserOnPostsListPageAndCanReportThatPost() {
+    final String postId = testsPostData.get(POST_ON_DESKTOP_ON_POSTS_LIST).getId();
+
+    PageWithPosts page = new PostsListPage().open();
+
+    assertThatPostReportedByOtherUserDoesNotHaveReportedIndicatorAndCanBeReportedByCurrentUser(page, postId);
+  }
+
+  @Test(groups = "discussions-loggedInUsersDesktopReporting",
+      dependsOnMethods = "userOnDesktopCanReportPostOnPostDetailsPage")
+  @Execute(asUser = User.USER_2)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
+  public void userOnDesktopCannotSeeReportedIndicatorOnPostsReportedByAnotherUserOnPostDetailsPageAndCanReportThatPost() {
+    final String postId = testsPostData.get(POST_ON_DESKTOP_ON_POST_DETAILS).getId();
+
+    PageWithPosts page = new PostDetailsPage().open(postId);
+
+    assertThatPostReportedByOtherUserDoesNotHaveReportedIndicatorAndCanBeReportedByCurrentUser(page, postId);
+  }
+
+  @Test(groups = "discussions-loggedInUsersDesktopReporting",
+      dependsOnMethods = "userOnDesktopCanReportPostOnUserPostsPage")
+  @Execute(asUser = User.USER_2)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
+  public void userOnDesktopCannotSeeReportedIndicatorOnPostsReportedByAnotherUserOnUserPostsPageAndCanReportThatPost() {
+    final PostEntity.Data postData = this.testsPostData.get(POST_ON_DESKTOP_ON_USER_PAGE);
+
+    PageWithPosts page = new UserPostsPage().open(postData.getAuthorId());
+
+    assertThatPostReportedByOtherUserDoesNotHaveReportedIndicatorAndCanBeReportedByCurrentUser(page, postData.getId());
+  }
+
+  // Third user on desktop
+
+  @Test(groups = "discussions-loggedInUsersDesktopReporting",
```

Comment:
Do we really need to be dependent on this all groups? Can we somehow work on testability here? Maybe there is a way to push this kind of a setup by using API?